keycloak.js

```diff
@@ -400,10 +400,19 @@ Keycloak.prototype.loginUrl = function (uuid, redirectUrl) {
   return url;
 };
 
-Keycloak.prototype.logoutUrl = function (redirectUrl) {
-  return this.config.realmUrl +
-  '/protocol/openid-connect/logout' +
-  '?redirect_uri=' + encodeURIComponent(redirectUrl);
+Keycloak.prototype.logoutUrl = function (redirectUrl, idTokenHint) {
+  let url = this.config.realmUrl +
+  '/protocol/openid-connect/logout';
+
+  if (idTokenHint) {
+    let urlParams = new URLSearchParams();
+    urlParams.append("id_token_hint", idTokenHint);
+    if (redirectUrl) {
+      urlParams.append("post_logout_redirect_uri", redirectUrl);
+    }
+    url += "?" + urlParams.toString();
+  }
+  return url;
 };
```

Comment:
Alternate proposal, we could construct a [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) and add the search params there. The only requirement would be that `config.realmUrl` has a valid structure, which means it needs at least a protocol and a host name.

```js
Keycloak.prototype.logoutUrl = function (redirectUrl, idTokenHint) {
  const url = new URL(this.config.realmUrl + '/protocol/openid-connect/logout');

  url.searchParams.set('post_logout_redirect_uri', redirectUrl);

  if (idTokenHint) {
    url.searchParams.set('id_token_hint', idTokenHint);
  }

  return url.toString();
};
```

Reply:
Ok, I rewrote that in this manner with the exception of the "post_logout_redirect_url" added just with the "id_token_hint" as mentioned above.

Note that the original version of this "logoutUrl" method did not used the URL/SearchParams and hence I've just continued with using it :-) But agree that this form with URL/SearchParams is slightly cleaner.